source/patterns/@aws-solutions-constructs/core/lib/glue-job-helper.ts

```diff
@@ -168,19 +182,22 @@ export function deployGlueJob(scope: Construct, glueJobProps: glue.CfnJobProps,
   };
 
   const _newGlueJobProps: glue.CfnJobProps = overrideProps(defaults.DefaultGlueJobProps(_jobRole!, glueJobProps,
-    _glueSecurityConfigName, _jobArgumentsList), glueJobProps);
-
-  let _scriptLocation: string;
-  if (isJobCommandProperty(_newGlueJobProps.command)) {
-    if (_newGlueJobProps.command.scriptLocation) {
-      _scriptLocation = _newGlueJobProps.command.scriptLocation;
-    } else {
-      throw Error('Script location has to be provided as an s3 Url location. Script location cannot be empty');
+    _glueSecurityConfigName, _jobArgumentsList, etlCodeAsset), glueJobProps);
+  if (etlCodeAsset) {
+    etlCodeAsset.grantRead(_jobRole);
+  } else {
+    let _scriptLocation: string;
+    if (isJobCommandProperty(_newGlueJobProps.command)) {
```

Comment:
This code gets hard to read with all the nested if statements. Let's make it easier to read by throwing errors immediately rather than nesting code inside of error checks. For example, this could be:

```
if (!isJobCommandProperty(_newGlueJobProps.command)) {
  throw Error('Invalid glueJobProps object provided');
}
```
This reduces the level of nesting by 1 overall. My next comment provides the ability to reduce more nesting I believe.

BTW - it seems that isJobCommandProperty() doesn't really validate the object - having at least 1 attribute doesn't make it a valid object, it needs to fully specify a job command doesn't it? Or are there defaults that I'm missing?

Reply:
1. updated the check for JobCommandProperty
2. Simplified loop as per recommendations.